### DISCUSSION TITLE
Fine tune ONNX Models (MMDetection) Inference for NVIDIA Jetson 调优目标检测 ONNX 通用模型，使其适合NVIDIA Jetson 平台部署

### DISCUSSION THREAD

**Discussion by OpenMMLab-Assistant-004** [open-mmlab/OpenMMLabCamp#562] [Other]
# Fine tune ONNX Models (MMDetection) Inference for NVIDIA Jetson 
 调优目标检测 ONNX 通用模型，使其适合NVIDIA Jetson 平台部署 

To deploy on the NVIDIA Jetson platform, you can follow these steps to optimize generic ONNX models to meet the specific requirements of the platform:
  Model Selection: Choose a generic [ONNX mode]( [[https://platform.openmmlab.com/deploee/onnx-list](https://platform.openmmlab.com/deploee/onnx-list)](https://platform.openmmlab.com/deploee/onnx-list](https://platform.openmmlab.com/deploee/onnx-list))  ) suitable for the Jetson platform as the base model.
  
 - Dataset : Collect a dataset relevant to your application scenario. For optimization tasks, you may need an annotated dataset for model fine-tuning.
 - Model Fine-tuning and Performance Optimization: Optimize the model on the Jetson platform to improve inference speed and efficiency. 
 - Deployment and Testing: Deploy the optimized model on the Jetson platform and ensure that the model performs as expected in real-world environments.
  
  Choose one of detection models to deploy:
 
 [MMDetection]([https://mmdeploy.readthedocs.io/en/latest/04-supported-codebases/mmdet.html#supported-models):](https://mmdeploy.readthedocs.io/en/latest/04-supported-codebases/mmdet.html#supported-models):)  rtmdet, yolov3, yolox
  
  Project requirements:
 - Submit the project's GitHub project link.
 - Provide a deployment tutorial that adheres to code standards and development requirements, including the following:
 - Introduce the dataset used for model optimization and testing, and explain the data preprocessing process.
 - Model Fine-tuning: Explain in detail the process of model fine-tuning, including the chosen base model, fine-tuning strategies, and performance optimization methods.
 - Model Deployment: Describe how to deploy the optimized model on the Jetson platform.
 - Summary and Evaluation: Summarize the project's achievements and evaluate the model's performance, efficiency, and adaptability on the Jetson platform.
 - Produce relevant test results and benchmarks after optimization.

请从 [MMDetection 目标检测模型]([https://mmdeploy.readthedocs.io/en/latest/04-supported-codebases/mmdet.html#supported-models)](https://mmdeploy.readthedocs.io/en/latest/04-supported-codebases/mmdet.html#supported-models))  中选择 1 个：[MMDetection]: rtmdet, yolov3, yolox mmseg: fastscnn, bisenetv2, stdc ，在 NVIDIA Jetson 平台上进行部署并调优模型，可通过以下几个步骤：
 - 模型选择：选择适合 Jetson 平台的通用 ONNX 模型作为基础模型。
 - 数据集准备：收集与您的应用场景相关的数据集。对于调优任务，您可能需要有标注的数据集，以便进行模型微调。
 - 模型微调，性能优化：在 Jetson 平台上，优化模型以提高推理速度和效率。
 - 部署和测试：将调优后的模型部署到 Jetson 平台上，确保模型在实际环境中的性能符合预期。 
  
 产出要求：
 1. 提交项目 Github 项目链接 
 2. 产出部署教程，符合代码规范和开发要求，包含以下内容：
 数据集和数据处理：介绍用于模型调优和测试的数据集，并解释数据的预处理过程。
 模型调优：详细说明模型调优的过程，包括选择的基础模型、微调策略和性能优化方法。
 模型部署：说明如何将调优后的模型部署到 Jetson 平台上 
 总结和评估：总结项目的成果，并评估模型在 Jetson 平台上的性能、效率和适应性。
 3. 对算法进行调优，产出相关测试结果和 benchmark
 
 硬件的使用可通过[部署平台]([https://platform.openmmlab.com/deploee?lang=zh-CN)](https://platform.openmmlab.com/deploee?lang=zh-CN)) ，使用 Jetson Orin

## Task Information

**Repository**：Other

**Difficulty**：Hard

**Points**：50

**Estimated Time Consumption**：21 days

## Task Status

**Claimant's Github ID**：jiongjiongli

**Claim Time**：2023年07月24日

**PR Link**：https://github.com/jiongjiongli/mmdet_jetson

## Points to Note

❗️请注意：此任务属于超级视客营任务，不能被领取！请登陆 [超级视客营平台](https://openmmlab.com/activity/codecamp) 按照流程申请任务，我们会尽快审核～

**Comment by jiongjiongli** [comment]
任务已完成，项目仓库：https://github.com/jiongjiongli/mmdet_jetson
